File: src/de/hsh/dennis/model/NpcLogic/SkinConfig.java
package de.hsh.dennis.model.NpcLogic;

import common.config.WindowConfig;
import javafx.scene.image.Image;

public class SkinConfig {

    public static class Level {
        public static Image level_Background = new Image("de/hsh/dennis/resources/background_Level.png");

        public static double speed = 1.0;

        public enum Difficulty {EASY, MEDIUM, HARD, NIGHTMARE, CUSTOM}
    }

    public static class Player {

        public static Image skin_standard = new Image("de/hsh/dennis/resources/actors/player/player_standard.png");

        public static Image skin_standard_right = new Image("de/hsh/dennis/resources/actors/player/player_standard_right.png");
        public static Image skin_standard_left = new Image("de/hsh/dennis/resources/actors/player/player_standard_left.png");

        public static Image skin_hit_left = new Image("de/hsh/dennis/resources/actors/player/player_hit_left.png");
        public static Image skin_hit_right = new Image("de/hsh/dennis/resources/actors/player/player_hit_right.png");

        public static Image skin_up_right = new Image("de/hsh/dennis/resources/actors/player/player_up_right.png");
        public static Image skin_up_left = new Image("de/hsh/dennis/resources/actors/player/player_up_left.png");

        public static Image skin_down_right = new Image("de/hsh/dennis/resources/actors/player/player_down_right.png");
        public static Image skin_down_left = new Image("de/hsh/dennis/resources/actors/player/player_down_left.png");


        public static double offsetX = 0;
        public static double offsetY = 100;

        public static double posX = (WindowConfig.window_width / 2.0) - ((int) skin_standard.getWidth() / 2.0) + offsetX;
        public static double posY = (WindowConfig.window_height / 2.0) - ((int) skin_standard.getHeight() / 2.0) + offsetY;

        public static double resetDelay = 0.2d;
    }

    public static class Bot {
        public static Image skin_standard = new Image("de/hsh/dennis/resources/actors/bot/bot.png");
    }

    public static class Hacker {
        public static Image skin_standard = new Image("de/hsh/dennis/resources/actors/hacker/hacker.png");
    }

    public static class Package {
        public static Image skin_standard = new Image("de/hsh/dennis/resources/actors/package/package.png");
        public static Image skin_healing = new Image("de/hsh/dennis/resources/actors/package/packageHealing.png");
    }
}

File: Games/Dennis/src/de/hsh/dennis/model/NpcLogic/SkinConfig.kt
package de.hsh.dennis.model.NpcLogic

import common.config.WindowConfig
import javafx.scene.image.Image

object SkinConfig {
    private const val resourceFolderPath = "/de/hsh/dennis/resources/"
    private const val skinFolder = resourceFolderPath + "skins/"

    object Level {
        var level_Background = Image(resourceFolderPath + "background_Level.png")
        var speed = 1.0

        enum class Difficulty {
            EASY, MEDIUM, HARD, NIGHTMARE, CUSTOM
        }
    }

    object Player {
        private const val playerPath = skinFolder + "player/"
        const val skin_standard_path = playerPath + "player_standard.png"
        const val skin_standard_right_path = playerPath + "player_standard_right.png"
        const val skin_standard_left_path = playerPath + "player_standard_left.png"
        const val skin_hit_left_path = playerPath + "player_hit_left.png"
        const val skin_hit_right_path = playerPath + "player_hit_right.png"
        const val skin_up_right_path = playerPath + "player_up_right.png"
        const val skin_up_left_path = playerPath + "player_up_left.png"
        const val skin_down_right_path = playerPath + "player_down_right.png"
        const val skin_down_left_path = playerPath + "player_down_left.png"
        var offsetX = 0.0
        var offsetY = 100.0
        const val skin_height = 200
        const val skin_width = 200
        val posX = WindowConfig.window_width / 2.0 - skin_width / 2.0 + offsetX
        val posY = WindowConfig.window_height / 2.0 - skin_width / 2.0 + offsetY
        var resetDelay = 0.2
    }

    object Bot {
        private const val botFolder = skinFolder + "bot/"
        const val skin_standard_path = botFolder + "bot.png"
        const val skin_width = 50
        const val skin_height = 50
    }

    object Hacker {
        private const val hackerFolder = skinFolder + "hacker/"
        const val skin_standard_path = hackerFolder + "hacker.png"
        const val skin_width = 50
        const val skin_height = 50
    }

    object Package {
        private const val packageFolder = skinFolder + "package/"
        const val skin_standard_path = packageFolder + "package.png"
        const val skin_healing_path = packageFolder + "packageHealing.png"
        const val skin_width = 50
        const val skin_height = 50
    }
}
